src/python/pants/backend/python/rules/pex.py

```diff
@@ -310,6 +331,16 @@ async def create_pex(
         *request.additional_args,
     ]
 
+    # NB: If `--platform` is specified, this signals that the PEX should not be built locally.
```

Comment:
Is that accurate? What if the requested `platform` happens to be the one we're running on?

Reply:
It's true enough for the purposes of the comment, I think. In John's words:

> Basically --platform implies remote / don't build

If it's `current`, then technically we are allowed to build wheels from any sdists. Otherwise, we must solely rely on bdists. So, generally, `--platform` does imply `--no-build`, outside of the special case of `--platform=current`.

--

Maybe, we should make this Pants logic more complex to inspect the platforms for `current`. I'm going to save that for a followup when John is back, though, as he understands this all better.

Reply:
I think that this also connects to https://docs.google.com/document/d/1hS0n6vp-hBxy4Xo-q9QA_elofzQtOubjfN4nSeIgp90/edit# ... based on the current design sketch there, if we're attempting to produce a binary for a particular target platform, it will need to be propagated in by the `binary` goal (or by the python rules?) as a param for the subgraph building the binary.